src/test/java/htsjdk/samtools/util/EdgeReadIteratorTest.java

```diff
@@ -398,14 +402,124 @@ public void testIntersectingIntervalWithComplicatedCigar() {
         assertEquals(21, locusPosition);
     }
 
+    /**
+     * Test for handling multiple intervals
+     */
+    @Test
+    public void testMultipleIntervals() {
+        String samHeaderString = "@HD\tSO:coordinate\tVN:1.0\n" +
+                "@SQ\tSN:chr1\tLN:100\n" +
+                "@SQ\tSN:chr2\tLN:100\n" +
+                "@SQ\tSN:chr3\tLN:100\n" +
+                "@SQ\tSN:chr4\tLN:100\n";
+
+        final SAMRecordSetBuilder builder = getRecordBuilder();
+        builder.setHeader(createSamFileHeader(samHeaderString));
+
+        builder.addFrag("fullyContainedInChr1", 0, 10, false, false, "10M10D10M", null, 10);
+        builder.addFrag("startOutsideChr2", 1, 15, false, false, "10M10D10M", null, 10);
+        builder.addFrag("endOutsideChr2", 1, 65, false, false, "10M10D10M", null, 10);
+        builder.addFrag("spanningThreeIntervalsChr4", 3, 1, false, false, "100M", null, 10);
+
+        IntervalList intervals = createIntervalList(samHeaderString +
+                "chr1\t1\t100\t+\ttest\n" +
+                "chr2\t20\t70\t+\ttest\n" +
+                "chr4\t20\t30\t+\ttest\n" +
+                "chr4\t40\t50\t+\ttest\n" +
+                "chr4\t60\t70\t+\ttest\n");
+
+        // These are the covered intervals that we expect to be covered
+        final Interval[] intervalsCovered = {
+                new Interval("chr1", 10, 19), // fullyContainedInChr1: first alignment block
+                new Interval("chr1", 30, 39), // fullyContainedInChr1: second alignment block
+                new Interval("chr2", 20, 24), // startOutsideChr2: first alignment block
+                new Interval("chr2", 35, 44), // startOutsideChr2: second alignment block
+                new Interval("chr2", 65, 70), // endOutsideChr2: first alignment block (second isn't covered at all)
+                new Interval("chr4", 20, 30), // spanningThreeIntervalsChr4: first interval
+                new Interval("chr4", 40, 50), // spanningThreeIntervalsChr4: second interval
+                new Interval("chr4", 60, 70), // spanningThreeIntervalsChr4: third interval
+        };
+
+        EdgeReadIterator iterator = new EdgeReadIterator(builder.getSamReader(), intervals);
+        iterator.iterator(); // We need this call to initialize the samReader. Usually a call to hasNext() would do that, which we don't need here.
```

Comment:
This is a bug here... next() should be valid without calling iterator() or hasNext().  Maybe we should fix that in AbstractLocusIterator.  I think there's a different bug where calling next() returns null instead of throwing NoSuchElementException.

Reply:
Fixed in AbstractLocusIterator